torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4547,6 +4547,16 @@ def _tensor(shape, dtype=dtype, low=None, high=None):
 
     return [SampleInput(tensor, args=args) for tensor, args in test_cases]
 
+def sample_inputs_one_hot(op_info, device, dtype, requires_grad, **kwargs):
+    def make_tensor_(shape, *, low, high):
+        return make_tensor(shape, device=device, dtype=dtype, low=low, high=high, requires_grad=requires_grad)
+
+    return [
+        SampleInput(make_tensor_((S,), low=0, high=10)),
+        SampleInput(make_tensor_((S,), low=0, high=5), kwargs=dict(num_classes=10)),
+        SampleInput(make_tensor_((L, M, S), low=0, high=10)),
+    ]
```

Comment:
> However, you can still make the cases disjoint from the `SampleInput` call if you find it reasonable.

I think that is only reasonable if we have more complicated sample generation. IMO your suggestions would make it more convoluted. Note that you missed the `low` / `high` arguments to `make_tensor` as well as the optional `kwargs` for the sample input. Ignoring the generator approach. a complete example would look like this:

```python
cases = (
    ((S,), 0, 10, None),
    ((S,), 0, 5, dict(num_classes=10)),
    ((L, M, S), 0, 10, None),
)

return [
    SampleInput(make_tensor_(shape, low=low, high=high), kwargs=kwargs)
    for shape, low, high, kwargs in cases
]
```

I think this is less readable than my approach, since you have no indication what each element in a `case` tuple does. One could (correctly) guess that the first element is a `shape` and the last is the operator `kwargs`, but the `int`'s in the middle? No chance unless you look at the implementation. 

IMO one should avoid a tuple of multiple different types in most situations. A named tuple would be solution that I could live with, but that seems a little over the top for such a simple example.